client/src/components/screens/PatientSubmission/index.jsx

```diff
@@ -132,8 +132,19 @@ function PatientSubmissionScreen(props) {
         };
 
         hasError = find(form.getFieldsError(), (o) => o.errors.length > 0);
-        if (values['analysis.tests']
-            && values['analysis.tests'].length > 0
+
+        const checkTest = () => {
+          if (values['analysis.tests']
+          && values['analysis.tests'].length > 0) {
+            if (values['analysis.tests'].includes(undefined)) {
+              return false;
+            }
+            return true;
```

Comment:
You should filter out the undefined entries from the array before checking `values['analysis.tests']`, and you'd also need to filter them out here https://github.com/Ferlab-Ste-Justine/clin-frontend/blob/master/client/src/components/screens/PatientSubmission/index.jsx#L341 too